### PR TITLE
Config canonical form

### PR DESCRIPTION
**Pull request by peterjcoffman**
This change causes all JSON configs to be standardized before saving in the DB.

I am not sure this is the fix we want, but it's worth discussing.

### CODE REVIEW

File: grails-app/controllers/io/xh/hoist/admin/ConfigAdminController.groovy [hunk 1]
```diff
@@ -26,7 +27,10 @@ class ConfigAdminController extends RestController {
     }
 
     protected void preprocessSubmit(JSONObject submit) {
+        def valueType = AppConfig.get(submit.id)?.valueType ?: submit.valueType
```
Comment (lbwexler) [inline grails-app/controllers/io/xh/hoist/admin/ConfigAdminController.groovy]: Does this re-order keys that would otherwise not be re-ordered?

Comment (peterjcoffman) [reply grails-app/controllers/io/xh/hoist/admin/ConfigAdminController.groovy]: Yes, it does. So this would probably be a no-go.

Comment (peterjcoffman) [reply grails-app/controllers/io/xh/hoist/admin/ConfigAdminController.groovy]: That's annoying. Nowadays the keys in a JS object have a meaningful order, but the JSON spec still says the keys in objects are unordered. So we have inconsistent behavior between JS and JSON.

**Comment by amcclain** [comment]
And closing this out as well - we can leave it to the config admin to determine the order of their keys, and how much they care about the indenting. We got the validation into place, and that's what was really important to us.

**Pull request closed without merging**